crates/ruma-events/src/room/beacon_info.rs

```diff
@@ -0,0 +1,81 @@
+//! Types for the `org.matrix.msc3489.beacon_info` state event, the unstable version of
+//! `m.beacon_info`.
+
+use std::time::Duration;
+
+use js_int::UInt;
+use ruma_common::{MilliSecondsSinceUnixEpoch, OwnedUserId};
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use crate::location::AssetContent;
+
+/// `BeaconInfoEventContent` is a struct that represents the content of a beacon_info state
+/// event. It contains information about a live location share.
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "org.matrix.msc3672.beacon_info", alias = "m.beacon_info", kind = State, state_key_type = OwnedUserId)]
+pub struct BeaconInfoEventContent {
+    /// The description of the location.
+    ///
+    /// It should be used to label the location on a map.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+
+    /// `live` is a boolean that should be true when a user starts sharing location.
+    pub live: bool,
+
+    /// The time when location sharing started, in milliseconds since the epoch.
```

Comment:
We don’t need to specify the unit, it’s clear from the type.

```suggestion
    /// The time when location sharing started.
```